plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueMetastoreModule.java

```diff
@@ -75,7 +84,33 @@ public Executor createExecutor(CatalogName catalogName, GlueHiveMetastoreConfig
             return directExecutor();
         }
         return new BoundedExecutor(
-                newCachedThreadPool(daemonThreadsNamed("hive-glue-%s")),
+                newCachedThreadPool(daemonThreadsNamed("hive-glue-partitions-%s")),
                 hiveConfig.getGetPartitionThreads());
     }
+
+    @Provides
+    @Singleton
+    @ForGlueColumnStatisticsRead
+    public Executor createStatisticsReadExecutor(CatalogName catalogName, GlueHiveMetastoreConfig hiveConfig)
+    {
+        if (hiveConfig.getReadStatisticsThreads() == 1) {
+            return directExecutor();
+        }
+        return new BoundedExecutor(
+                newCachedThreadPool(daemonThreadsNamed("hive-glue-statistics-read-%s")),
+                hiveConfig.getReadStatisticsThreads());
```

Comment:
extract helper method to create executor given number of threads and name. All three look exactly the same.